terrapower/physics/neutronics/dragon/dragonWriter.py

```diff
@@ -70,131 +84,79 @@ def _readTemplate(self):
-            # requires some engineering judgment, and this structure allows template
-            # creators to apply their own judgment.
-            # There is some basic data that templates filter out of
-            # self shielding (SS) calculation, or apply different inrs to nuclides.
-            selfShieldingFilterData = {
-                # Sometimes very low density components must be filtered from
-                # self shielding (by not setting inrs) for the SS to run.
-                "mixtureMassDensity": mixtureMassDensity,
-                # Heavy metals should almost always have SS.
-                "heavyMetal": nuclideBase.isHeavyMetal(),
-                # Sometimes it makes sense to filter out minor nuclides from SS.
-                "atomFrac": nDens / totalADensityModeled,
+class DragonWriterHomogenized(DragonWriter):
+    """
+    Write DRAGON inputs with homogenized compositions.
+
+    This subclass assumes that the DRAGON case will represent one or 
+    more armi objects.
+
+    The current implementation is capable of writing MIX cards for multiple
+    compositions at once but does not yet have the ability to write geometry
+    representation for anything beyond 0-D.
+    """
+
+    def _buildTemplateData(self):
+        templateData = DragonWriter._buildTemplateData(self)
+
+        templateData.update(
+            {
+                "mixtures": self._makeMixtures(),
+                "buckling": self.options.xsSettings.criticalBuckling,
             }
+        )
+        return templateData
 
-            # This density may be 0.0, and DRAGON will run fine.
-            nDensCardData.append(
-                (
-                    f"{nuclideBase.label}{self.options.xsID}",
-                    dragonId,
-                    nDens,
-                    selfShieldingFilterData,
-                )
-            )
-        return nDensCardData
+    def _makeMixtures(self):
+        """Make a DragonMixture from each object slated for inclusion in the input."""
+        return [DragonMixture(obj, self.options) for obj in self.armiObjs]
 
-    @staticmethod
-    def getDragLibNucID(nucBase):
-        """
-        Return the DRAGLIB isotope name for this nuclide.
-        
-        Parameters
-        ----------
-        nucBase : NuclideBase
-            The nuclide to get the DRAGLIB ID for.
 
-        Notes
-        -----
-        These IDs are compatible with DRAGLIB nuclear data format which is available:
-        https://www.polymtl.ca/merlin/libraries.htm
-        """
-        metastable = nucBase.state
-        # DRAGON is case sensitive on nuc names so lower().capitalize() matters.
-        dragLibId = f"{nucBase.element.symbol.lower().capitalize()}{nucBase.a}"
-        if metastable > 0:
-            # Am242m, etc
-            dragLibId += "m"
-        return dragLibId
-
-    def _getFuelTempInK(self):
+class DragonMixture:
+    """
+    Data structure for a single mixture in Dragon.
+
+    Each mixture has:
```

Comment:
Its a little weird that it says it has it but its not stored.

Maybe each mixture can determine its:
        * Temperature
        * Number density vector
        * Mapping between library names and internal nuclide names
        * Self-shielding

Reply:
changed to "is associated with"